### PR TITLE
[macOS] Fix crash when setting window properties

### LINKED ISSUE #21306: [macOS] Crash in multi-window mode in the Controls.sample project
### Description

Setting window size crashes the MAUI application.

### Steps to Reproduce

Compile and run `Controls.Sample` project:

```bash
cd src/Controls/samples/Controls.Sample
dotnet build -t:run --configuration Debug -p:RuntimeIdentifier=maccatalyst-arm64 --framework net8.0-maccatalyst
```

In the app, navigate to Core -> Multi-Window and click `Set window size to 700x500` and the application crashes.

### Link to public reproduction project repository

_No response_

### Version with bug

Nightly / CI build (93940d6f931a54bce074f8dfd2013440e0827cf3). 

However, I can see something similar in my MAUI app as well but reproducing the issue in the Controls.Sample project is easiest so I reported it this way.

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

macOS

### Affected platform versions

Apple Mac M1 Sonoma 14.4

### Did you find any workaround?

No.

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by MartyIX**
### Description of Change

I'm not really sure if the approach of the patch is correct. It's the only one I could make as I don't understand details here. 

Feedback is welcome.

See https://github.com/dotnet/maui/issues/21306#issuecomment-2007580475 for an explanation of the bug. 

### Issues Fixed

Fixes #21306

### CODE REVIEW

File: src/Controls/src/Core/Window/Window.cs [hunk 1]
```diff
@@ -198,6 +198,10 @@ public double MinimumHeight
 			return ValidatePositive(coord);
 		}
 
+#if MACCATALYST
+		bool _frameUpdateInProgress = false;
```
Comment (mattleibow) [inline src/Controls/src/Core/Window/Window.cs:202]: Is there a way to use the `_batchFrameUpdate` field? That is already being set when the frame is updating.

File: src/Controls/src/Core/Window/Window.cs [hunk 2]
```diff
@@ -206,9 +210,26 @@ void IWindow.FrameChanged(Rect frame)
...
+#if MACCATALYST
+			// MacCatalyst does not support setting window properties: X, Y, width, and height.
+			if (_frameUpdateInProgress) {
+				X = frame.X;
+				Y = frame.Y;
+				Width = frame.Width;
+				Height = frame.Height;
+
+				return;
+			}
```
Comment (mattleibow) [inline src/Controls/src/Core/Window/Window.cs:223]: What is the difference between setting this and using `SetValueCore`?

**Comment by PureWeen** [comment]
Closing this PR in favor of
https://github.com/dotnet/maui/pull/22413

Which pretty has the same end result.

Once we merge 22413, if that doesn't appear to fix @MartyIX let me know!

**Pull request closed without merging**